diff --git a/hw/core-v-mini-mcu/core_v_mini_mcu.sv.tpl b/hw/core-v-mini-mcu/core_v_mini_mcu.sv.tpl
index 8bff6e0..7033b07 100644
--- a/hw/core-v-mini-mcu/core_v_mini_mcu.sv.tpl
+++ b/hw/core-v-mini-mcu/core_v_mini_mcu.sv.tpl
@@ -43,10 +43,14 @@ module core_v_mini_mcu
 
     output logic cpu_subsystem_powergate_switch_o,
     input  logic cpu_subsystem_powergate_switch_ack_i,
+    output logic cpu_subsystem_sleep_o,
     output logic peripheral_subsystem_powergate_switch_o,
     input  logic peripheral_subsystem_powergate_switch_ack_i,
+    output logic peripheral_subsystem_clkgate_en_o,
     output logic [core_v_mini_mcu_pkg::NUM_BANKS-1:0] memory_subsystem_banks_powergate_switch_o,
     input  logic [core_v_mini_mcu_pkg::NUM_BANKS-1:0] memory_subsystem_banks_powergate_switch_ack_i,
+    output logic [core_v_mini_mcu_pkg::NUM_BANKS-1:0] memory_subsystem_banks_set_retentive_o,
+    output logic [core_v_mini_mcu_pkg::NUM_BANKS-1:0] memory_subsystem_clkgate_en_o,
     output logic [core_v_mini_mcu_pkg::EXTERNAL_DOMAINS-1:0] external_subsystem_powergate_switch_o,
     input  logic [core_v_mini_mcu_pkg::EXTERNAL_DOMAINS-1:0] external_subsystem_powergate_switch_ack_i,
     output logic [core_v_mini_mcu_pkg::EXTERNAL_DOMAINS-1:0] external_subsystem_powergate_iso_o,
@@ -177,6 +181,11 @@ module core_v_mini_mcu
     rv_timer_intr[1]
   };
 
+  assign cpu_subsystem_sleep_o = core_sleep;
+  assign memory_subsystem_banks_set_retentive_o = memory_subsystem_banks_set_retentive;
+  assign memory_subsystem_clkgate_en_o = memory_subsystem_clkgate_en;
+  assign peripheral_subsystem_clkgate_en_o = peripheral_subsystem_clkgate_en;
+
   cpu_subsystem #(
       .BOOT_ADDR(BOOT_ADDR),
       .PULP_XPULP(PULP_XPULP),
