diff --git a/hw/fpga/prim_xilinx_clk.sv b/hw/fpga/prim_xilinx_clk.sv
index 0713f2d..9a7411d 100644
--- a/hw/fpga/prim_xilinx_clk.sv
+++ b/hw/fpga/prim_xilinx_clk.sv
@@ -76,40 +76,6 @@ module pulp_clock_inverter (
 
 endmodule
 
-module cv32e40p_clock_gate (
-    input  logic clk_i,
-    input  logic en_i,
-    input  logic scan_cg_en_i,
-    output logic clk_o
-);
-
-  xilinx_clk_gating clk_gate_i (
-      .clk_i,
-      .en_i,
-      .test_en_i(scan_cg_en_i),
-      .clk_o
-  );
-
-endmodule
-
-module cv32e40x_clock_gate #(
-    parameter LIB = 0
-) (
-    input  logic clk_i,
-    input  logic en_i,
-    input  logic scan_cg_en_i,
-    output logic clk_o
-);
-
-  xilinx_clk_gating clk_gate_i (
-      .clk_i,
-      .en_i,
-      .test_en_i(scan_cg_en_i),
-      .clk_o
-  );
-
-endmodule
-
 module tc_clk_gating (
     input  logic clk_i,
     input  logic en_i,
